core/src/main/java/org/apache/iceberg/hadoop/HadoopTableOperations.java

```diff
@@ -353,8 +346,12 @@ int findVersion() {
    * @param src the source file
    * @param dst the destination file
    */
-  private void renameToFinal(FileSystem fs, Path src, Path dst) {
+  private void renameToFinal(FileSystem fs, Path src, Path dst, int nextVersion) {
     try {
+      lockManager.acquire(dst.toString(), Thread.currentThread().getName());
```

Comment:
The current thread name can repeat across processes. This should be something unique. The `GlueCatalog` uses the metadata file to commit. I think this should do the same and use `src`.